prov/efa/src/rdm/efa_rdm_srx.c

```diff
@@ -42,12 +87,46 @@
  * @param[in] addr the source address of the incoming message
  * @param[in] size the size of the incoming message
  * @param[out] peer_rx_entry the obtained peer_rx_entry
- * @return int 0 on success, a negative integer on failure
+ * @return int FI_SUCCESS when a matched rx entry is found, -FI_ENOENT when the
+ * match is not found, other negative integer for other errors.
  */
 static int efa_rdm_srx_get_msg(struct fid_peer_srx *srx, fi_addr_t addr,
 		       size_t size, struct fi_peer_rx_entry **peer_rx_entry)
 {
-    return -FI_ENOSYS;
+	struct rxr_ep *rxr_ep;
+	struct rxr_pkt_entry *rx_pkt_entry;
+	struct rxr_op_entry *rx_entry;
+	int ret;
+
+	rxr_ep = (struct rxr_ep *) srx->ep_fid.fid.context;
+	/*
+	 * TODO:
+	 * In theory we should not need to allocate a pkt_entry to get a rx entry,
+	 * but the current EFA code needs an pkt_entry as input. A major refactor
+	 * is needed to split the context (addr, size etc.) and data from pkt entry
+	 * and make rxr_*_get_*_rx_entry needs context only.
+	 */
+	rx_pkt_entry = rxr_pkt_entry_alloc(rxr_ep, rxr_ep->efa_rx_pkt_pool, RXR_PKT_FROM_EFA_RX_POOL);
+	if (OFI_UNLIKELY(!rx_pkt_entry)) {
+		EFA_WARN(FI_LOG_EP_CTRL,
+			"Unable to allocate rx_pkt_entry\n");
+		return -FI_ENOMEM;
+	}
+	efa_rdm_srx_construct_pkt_entry(rxr_ep, addr, size, 0, ofi_op_msg, rx_pkt_entry);
+
+	rx_entry = rxr_pkt_get_msgrtm_rx_entry(rxr_ep, &rx_pkt_entry);
```

Comment:
I was confused why it's OK to always call `rxr_pkt_get_msgrtm_rx_entry` even though it has `msgrtm` in its name.

But `rxr_pkt_get_msgrtm_rx_entry` is the only function where we search the `rx_list` in `prov/efa`. For other types of messages, we either don't need rx_entry (e.g. handshake packets) or we already have it (e.g. data packets).